### PR TITLE
update sailor 267

### PR DESCRIPTION
**Pull request by kirill-levitskiy**

### CODE REVIEW

File: lib/actions/pub.js [hunk 1]
```diff
@@ -1,65 +1,56 @@
...
     };
-    var data = messages.newMessageWithBody(body);
+    const data = messages.newMessageWithBody(body);
     self.emit('data', data);
-  }
-
-  function emitError(e) {
-    console.log('Oops! Error occurred');
+  }).catch((e) => {
+    this.logger.info('Oops! Error occurred');
```
Comment (shulkaolka) [inline lib/actions/pub.js]: await and then shouldn't be used together.
as an example, you can use:
```
let messageId;
    try{
      messageId = await topic.publish(bufferData)
    } catch (e) {
      this.logger.info('Oops! Error occurred');
      self.emit('error', e);
    }
    this.logger.info('Message sent with ID:', messageId);
    body = {
      messageID: messageId,
    }
    const data = messages.newMessageWithBody(body);
    self.emit('data', data);
```

File: lib/triggers/pull.js [hunk 2]
```diff
@@ -1,112 +1,79 @@
...
+      projectId,
+      subscriptionName,
+    );
 
-  function pullSubscription() {
-    console.log(`Polling from subscription ${subscription.name}`);
-    return Q.ninvoke(subscription, 'pull', {
-      returnImmediately: true,
-      maxMessages: 1000
+    await subClient.getSubscription({ subscription: formattedSubscription }).catch(async (err) => {
```
Comment (shulkaolka) [inline lib/triggers/pull.js:34]: please use `try..catch` for async function

File: package.json [hunk 3]
```diff
@@ -19,25 +19,31 @@
...
-  "eslintConfig": {
-    "extends": "xo-space",
-    "env": {
-      "mocha": true
-    }
+  "dependencies": {
+    "elasticio-node": "0.0.9",
+    "elasticio-sailor-nodejs": "2.6.7",
+    "@google-cloud/pubsub": "1.7.3",
+    "q": "1.4.1",
```
Comment (shulkaolka) [inline package.json:33]: maybe we don't use this lib and can delete?

File: spec/verifyCredentials.spec.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    }
+  });
+
+  const cfg = {
+    projectId: 'projectId',
+    client_email: 'client@email.com',
+    private_key: 'private_key',
+  };
+
+  it('should fail to validate an invalid credentials', async () => {
```
Comment (shulkaolka) [inline spec/verifyCredentials.spec.js:28]: please, add test for a successful case

File: verifyCredentials.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable camelcase */
+const { PubSub } = require('@google-cloud/pubsub');
+const logger = require('@elastic.io/component-logger')();
```
Comment (shulkaolka) [inline verifyCredentials.js:3]: verify credentials function supports this.logger. Please replace it.

Comment (kirill-levitskiy) [reply verifyCredentials.js]: replaced

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
+## 1.7.0 (May 12, 2020)
```
Comment (shulkaolka) [inline CHANGELOG.md:1]: According to changes my suggestion is up version to 2.0.0. Or all this changes are backward compatible?

**Pull request merged**